src/main/java/org/terasology/logic/inventory/StartingInventorySystem.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.logic.inventory;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.players.event.OnPlayerSpawnedEvent;
+import org.terasology.registry.In;
+import org.terasology.utilities.Assets;
+import org.terasology.world.block.BlockManager;
+import org.terasology.world.block.family.BlockFamily;
+import org.terasology.world.block.items.BlockItemFactory;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class StartingInventorySystem extends BaseComponentSystem {
+
+    private static final Logger logger = LoggerFactory.getLogger(StartingInventorySystem.class);
+
+    @In
+    BlockManager blockManager;
+
+    @In
+    InventoryManager inventoryManager;
+
+    @In
+    EntityManager entityManager;
+
+    @In
+    PrefabManager prefabManager;
+
+    BlockItemFactory blockFactory;
+
+    @Override
+    public void initialise() {
+        blockFactory = new BlockItemFactory(entityManager);
+    }
+
+    @ReceiveEvent(components = {StartingInventoryComponent.class, InventoryComponent.class})
+    public void onStartingInventory(OnPlayerSpawnedEvent event, EntityRef entityRef) {
+        StartingInventoryComponent startingInventoryComponent =
+                entityRef.getComponent(StartingInventoryComponent.class);
+        if (!startingInventoryComponent.provided) {
+            InventoryComponent inventoryComponent = entityRef.getComponent(InventoryComponent.class);
+            if (entityRef.getParentPrefab() != null) {
+                logger.info("Adding starting inventory to {}, entity has {} slots",
+                        entityRef.getParentPrefab().getName(), inventoryComponent.itemSlots.size());
+            }
+            for (StartingInventoryComponent.InventoryItem item : startingInventoryComponent.items) {
+                addToInventory(entityRef, item, inventoryComponent);
+            }
+            startingInventoryComponent.provided = true;
```

Comment:
The `provided` field seems to be a flag to indicate that the items have already been given?
Have you considered just removing the StartingInventoryComponent?

Reply:
Ah yes, of course. Oops, should've thought of that. Thanks :)